crates/gitbutler-repo/src/commands.rs

```diff
@@ -59,51 +74,61 @@ impl RepoCommands for Project {
             );
         }
 
+        let status = ctx.repository().status_file(relative_path)?;
+
+        if status.is_wt_new() {
```

Comment:
GitButler may show worktree files, which may be 

* untracked
* modified
* deleted in the worktree
* deleted in the worktree and index

Semantically this boils down to trying to read, in order from top to bottom:

* worktree
* index
* `HEAD^{tree}

It may also show files that are already committed, which is when they will always be contained in trees. The UI will only display added, deleted or modified ones

* deleted files have to be fetched from the 'previous' tree
* added files are found in the 'current' tree
* modified files can also be found in the `current` tree

In this case, we'd not want to look at the worktree or the index at all, but the UI would probably have to decide which tree to get the relative path from.

The semantics of `read_file_from_workspace()` fit better to the `tree` version (the latter example), and could rather be `read_file_from_tree(tree_id, relative_path)`, while a new function could handle the former case.

CC @Caleb-T-Owens and @ndom91 .

Reply:
Thank you for the clear explanation. Based on my understanding, it seems like it would be better to separate the part where `read_file_from_workspace` reads `HEAD^{tree}` into a function like `read_file_from_tree(tree_id, relative_path)`.

Additionally, I will work on handling the logic for modified or deleted files in the worktree or `HEAD^{tree}` within `read_file_from_workspace`.

Reply:
Based on the review, I have applied a few changes:

- In `read_file_from_workspace`, I attempted to handle file lookups in the order of worktree, index, and `HEAD^{tree}`.
- Separated the logic for tree lookups in `read_file_from_workspace` into a new function `read_file_from_tree(tree_id, relative_path)`.
- Added conditions to handle the untracked, modified, and deleted states that GitButler displays.

### Modified

https://github.com/user-attachments/assets/4d6f1df7-766e-4807-9e58-9330dd8b4120

### Removed

https://github.com/user-attachments/assets/476d358d-6f00-447a-85e5-3745a4cd20b6


P.S.: In the UI worktree, the file states (delete, add) are clearly visible, but in the committed tree, only the modified state appears. This can be a bit confusing.